core/src/main/scala/com/salesforce/op/filters/Summary.scala

```diff
@@ -64,3 +75,162 @@ case object Summary {
+  }
+  private[this] val minOp: BinaryOperator[Double] = new BinaryOperator[Double] with Serializable {
+    def apply(s: Double, t: Double): Double = math.min(s, t)
+  }
+  private[this] val sumOp: BinaryOperator[Double] = new BinaryOperator[Double] with Serializable {
+    def apply(s: Double, t: Double): Double = s + t
+  }
+
+  final def getCount(): Double = count.get
+
+  final def getDistribution(): Array[(Double, Double)] = distribution.toArray.sortBy(_._1)
+
+  final def getFeatureDistribution(
+    featureKey: FeatureKey,
+    totalCount: Double,
+    featureDistributionType: FeatureDistributionType): FeatureDistribution = {
+    val thisCount = getCount
+    val nullCount = totalCount - thisCount
+    val dist = getDistribution
+
+    FeatureDistribution(
+      name = featureKey._1,
+      key = featureKey._2,
+      count = totalCount.toLong,
+      nulls = nullCount.toLong,
+      distribution = dist.map(_._2),
+      summaryInfo = dist.map(_._1),
+      `type` = featureDistributionType)
+  }
+
+  final def getNumTokens(): Double = numTokens.get
+
+  final def getMinTokens(): Double = minTokens.get
+
+  final def getMaxTokens(): Double = maxTokens.get
+
+  final def merge(other: TextSummary): this.type = synchronized {
+    count.accumulateAndGet(other.getCount, sumOp)
+    maxTokens.accumulateAndGet(other.getMaxTokens, maxOp)
+    minTokens.accumulateAndGet(other.getMinTokens, minOp)
+    numTokens.accumulateAndGet(other.getNumTokens, sumOp)
+
+    this
+  }
+
+  final def mergeDistribution(other: TextSummary): this.type = {
+    other.getDistribution.foreach { case (point, count) =>
+      val newCount = distribution.get(point).getOrElse(0.0) + count
+      distribution += (point -> newCount)
+    }
+
+    this
+  }
+
+  final def setHashingTF(): this.type = synchronized {
+    hashingTFOpt = Option {
+      new HashingTF(numFeatures = textFormula(this)).setBinary(false)
+        .setHashAlgorithm(HashAlgorithm.MurMur3.entryName.toLowerCase)
+    }
+
+    this
+  }
+
+  final def update(text: Seq[String]): this.type = synchronized {
+    val size: Double = text.length
+    count.accumulateAndGet(1.0, sumOp)
+    maxTokens.accumulateAndGet(size, maxOp)
+    minTokens.accumulateAndGet(size, minOp)
+    numTokens.accumulateAndGet(size, sumOp)
+
+    this
+  }
+
+  final def updateDistribution(text: Seq[String]): this.type = synchronized {
+    hashingTFOpt match {
+      case Some(hashingTF) =>
+        hashingTF.transform(text).toArray.zipWithIndex.foldLeft(()) { case (_, (count, idx)) =>
+          val newCount = distribution.get(idx).getOrElse(1.0) + count
+          distribution += (idx.toDouble -> newCount)
+        }
+      case None =>
+        throw new RuntimeException("HashingTF must be set in order to update text summary distribution")
+    }
+
+    this
+  }
+}
+
+object TextSummary {
+  implicit val semigroup = new Semigroup[TextSummary] {
+    def plus(l: TextSummary, r: TextSummary): TextSummary = l.merge(r)
+  }
+}
+
```

Comment:
could you add some comments for the newly added classes, is this for all numerics? (compared to textSummary)?